File: courtlistener/api/originating_court_information.py
"""
Originating Court Information API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class OriginatingCourtInformationAPI(BaseAPI):
    """API for accessing originating court information."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "originating-court-information/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List originating court information records.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of originating court information
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, info_id: int) -> Dict[str, Any]:
        """
        Get a specific originating court information by ID.
        
        Args:
            info_id: Originating court information ID
        
        Returns:
            Originating court information data
        """
        return self.client.get(f"{self.endpoint}{info_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated originating court information.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for originating court information
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/OriginatingCourtInformation.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * OriginatingCourtInformation API endpoint
 */
class OriginatingCourtInformation extends BaseApi
{
    protected string $endpoint = 'originating-court-information/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List originating court information with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listOriginatingCourtInformation(array $params = []): array
    {
        return $this->list($params);
    }

    /**
     * Get a specific originating court information by ID
     *
     * @param int|string $id OriginatingCourtInformation ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getOriginatingCourtInformation($id, array $params = []): array
    {
        return $this->get($id, $params);
    }

    /**
     * Search originating court information
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchOriginatingCourtInformation(array $params = []): array
    {
        return $this->search($params);
    }
}
